resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -130,12 +159,28 @@ private[spark] class KubernetesClusterSchedulerBackend(
     super.start()
     executorWatchResource.set(kubernetesClient.pods().withLabel(SPARK_APP_ID_LABEL, applicationId())
       .watch(new ExecutorPodsWatcher()))
+
+    allocator = new KubernetesAllocator(kubernetesClient)
+    allocator.start()
+
     if (!Utils.isDynamicAllocationEnabled(sc.conf)) {
       doRequestTotalExecutors(initialExecutors)
+    } else {
+      shufflePodCache = shuffleServiceConfig
+        .map { config =>
+          Some(new ShufflePodCache(kubernetesClient,
+            config.shuffleNamespace,
+            config.shuffleLabels))
+        }.getOrElse(None)
```

Comment:
Just use `.map(config => new ShufflePodCache)`. There's no need to wrap the return value from the `map` in an `Option` nor to use `getOrElse`.